actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1741,7 +1741,7 @@ def name_for_action(as, action) #:nodoc:
 
             name = @scope.action_name(name_prefix, prefix, collection_name, member_name)
 
-            if candidate = name.compact.join("_").presence
+            if candidate = name.select(&:present?).join("_").presence
```

Comment:
@rono23 maybe we can rename `name` to `action_name` to refer to what it actually stands for, wdyt?